kong/client.go

```diff
@@ -164,6 +165,34 @@ func NewClient(baseURL *string, client *http.Client) (*Client, error) {
 	return kong, nil
 }
 
+// WithWorkspace return the client with a defined workspace
+func (c *Client) WithWorkspace(workspace string) *Client {
+	c.SetWorkspace(workspace)
+	return c
+}
+
```

Comment:
This function modifies the object it's being called on and, as such, is not different than just calling `SetWorkspace`. I would be misled into thinking that it creates a copy without modifying the receiver.

Why is this function needed? I think that implementing it the right way will likely be cumbersome. Suggestion: remove it altogether.